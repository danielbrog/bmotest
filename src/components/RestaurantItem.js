import React from "react";

const RestaurantItem = (props) => {
  const { restaurant } = props;

  const price = new Array(restaurant.price);
  price.fill("$").join("");

  return (
    <div className="restaurant__info">
      <div className="restaurant__name">
        <div className="restaurant__price">{price}</div>
        {restaurant.name}
      </div>
      <div className="restaurant__address">{restaurant.address}</div>
      <div className="restaurant__area">{restaurant.area}</div>
      <div className="restaurant__phone">{restaurant.phone}</div>
    </div>
  );
};

export default RestaurantItem;
