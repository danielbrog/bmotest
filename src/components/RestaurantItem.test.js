import React from "react";
import RestaurantItem from "./RestaurantItem";
import renderer from "react-test-renderer";
import mockRestaurant from "../mocks/restaurant.json";

test("RestaurantItem renders", () => {
  const testRenderer = renderer
    .create(<RestaurantItem restaurant={mockRestaurant} />)
    .toJSON();
  expect(testRenderer).toMatchSnapshot();
});
