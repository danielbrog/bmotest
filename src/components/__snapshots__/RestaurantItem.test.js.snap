// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantItem renders 1`] = `
<div
  className="restaurant__info"
>
  <div
    className="restaurant__name"
  >
    <div
      className="restaurant__price"
    >
      $
      $
      $
      $
    </div>
    Scaramouche Restaurant
  </div>
  <div
    className="restaurant__address"
  >
    1 Benvenuto Place
  </div>
  <div
    className="restaurant__area"
  >
    Toronto / SW Ontario
  </div>
  <div
    className="restaurant__phone"
  >
    4169618011
  </div>
</div>
`;
